weather_dl/download_pipeline/partition_test.py

```diff
@@ -118,21 +121,26 @@ def test_partition_multi_params_multi_key(self):
             }
         }
 
-        actual = self.create_partition_configs(config)
+        config_obj = Config.from_config(config)
+        actual = self.create_partition_configs(config_obj)
 
         expected = [
-            {'parameters': dict(config['parameters'], api_key='KKKK1', api_url='UUUU1', __subsection__='research'),
-             'selection': {**config['selection'],
-                           **{'year': ['2015'], 'month': ['1']}}},
-            {'parameters': dict(config['parameters'], api_key='KKKK2', api_url='UUUU2', __subsection__='cloud'),
-             'selection': {**config['selection'],
-                           **{'year': ['2015'], 'month': ['2']}}},
-            {'parameters': dict(config['parameters'], api_key='KKKK3', api_url='UUUU3', __subsection__='deepmind'),
-             'selection': {**config['selection'],
-                           **{'year': ['2016'], 'month': ['1']}}},
-            {'parameters': dict(config['parameters'], api_key='KKKK1', api_url='UUUU1', __subsection__='research'),
-             'selection': {**config['selection'],
-                           **{'year': ['2016'], 'month': ['2']}}},
+            Config.from_config(
+                {'parameters': dict(config['parameters'], api_key='KKKK1', api_url='UUUU1', subsection_name='research'),
+                 'selection': {**config['selection'],
+                               **{'year': ['2015'], 'month': ['1']}}}),
+            Config.from_config(
+                {'parameters': dict(config['parameters'], api_key='KKKK2', api_url='UUUU2', subsection_name='cloud'),
+                 'selection': {**config['selection'],
+                               **{'year': ['2015'], 'month': ['2']}}}),
+            Config.from_config(
+                {'parameters': dict(config['parameters'], api_key='KKKK3', api_url='UUUU3', subsection_name='deepmind'),
+                 'selection': {**config['selection'],
+                               **{'year': ['2016'], 'month': ['1']}}}),
+            Config.from_config(
+                {'parameters': dict(config['parameters'], api_key='KKKK1', api_url='UUUU1', subsection_name='research'),
+                 'selection': {**config['selection'],
+                               **{'year': ['2016'], 'month': ['2']}}}),
```

Comment:
What about using a list comprehension here? Something like: 

```suggestion
            Config.from_config(it) for it in [
                {'parameters': dict(config['parameters'], api_key='KKKK1', api_url='UUUU1', subsection_name='research'),
                 'selection': {**config['selection'],
                               **{'year': ['2015'], 'month': ['1']}}},
                {'parameters': dict(config['parameters'], api_key='KKKK2', api_url='UUUU2', subsection_name='cloud'),
                 'selection': {**config['selection'],
                               **{'year': ['2015'], 'month': ['2']}}},
                {'parameters': dict(config['parameters'], api_key='KKKK3', api_url='UUUU3', subsection_name='deepmind'),
                 'selection': {**config['selection'],
                               **{'year': ['2016'], 'month': ['1']}}},
                {'parameters': dict(config['parameters'], api_key='KKKK1', api_url='UUUU1', subsection_name='research'),
                 'selection': {**config['selection'],
                               **{'year': ['2016'], 'month': ['2']}}},
            ]                   
```